pkg/securitypolicy/securitypolicyenforcer.go

```diff
@@ -269,691 +223,33 @@ func CreateSecurityPolicyEnforcer(
-// the enforcement, the method also narrows down possible container IDs with
-// the same overlay.
-func (pe *StandardSecurityPolicyEnforcer) enforceMountPolicy(sandboxID, containerID string, mounts []oci.Mount) (err error) {
-	possibleIndices := pe.possibleIndicesForID(containerID)
-
-	for _, mount := range mounts {
-		// first check against default mounts
-		if err := pe.enforceDefaultMounts(mount); err == nil {
-			continue
-		}
-
-		mountOk := false
-		// check against user provided mount constraints, which helps to figure
-		// out which container this mount spec corresponds to.
-		for _, pIndex := range possibleIndices {
-			cont := pe.Containers[pIndex]
-			if err = cont.matchMount(sandboxID, mount); err == nil {
-				mountOk = true
-			} else {
-				pe.narrowMatchesForContainerIndex(pIndex, containerID)
-			}
-		}
-
-		if !mountOk {
-			retErr := fmt.Errorf("mount %+v is not allowed by mount constraints", mount)
-			return retErr
-		}
-	}
-
-	return nil
 }
 
-// validate checks given OCI mount against mount policy. Destination is checked
-// by direct string comparisons and Source is checked via a regular expression.
-// This is done this way, because container path (Destination) is always fixed,
-// however, the host/UVM path (Source) can include IDs generated at runtime and
-// impossible to know in advance.
-//
-// NOTE: Different matching strategies can be added by introducing a separate
-// path matching config, which isn't needed at the moment.
-func (m *mountInternal) validate(mSpec oci.Mount) error {
-	if m.Type != mSpec.Type {
-		return fmt.Errorf("mount type not allowed by policy: expected=%q, actual=%q", m.Type, mSpec.Type)
-	}
-	if ok, _ := regexp.MatchString(m.Source, mSpec.Source); !ok {
-		return fmt.Errorf("mount source not allowed by policy: expected=%q, actual=%q", m.Source, mSpec.Source)
-	}
-	if m.Destination != mSpec.Destination && m.Destination != "" {
-		return fmt.Errorf("mount destination not allowed by policy: expected=%q, actual=%q", m.Destination, mSpec.Destination)
-	}
-	if !stringSlicesEqual(m.Options, mSpec.Options) {
-		return fmt.Errorf("mount options not allowed by policy: expected=%q, actual=%q", m.Options, mSpec.Options)
-	}
-	return nil
-}
-
-// matchMount matches given OCI mount against mount constraints. If no match
-// found, the mount is not allowed.
-func (c *securityPolicyContainer) matchMount(sandboxID string, m oci.Mount) (err error) {
-	for _, constraint := range c.Mounts {
-		// now that we know the sandboxID we can get the actual path for
-		// various destination path types by adding a UVM mount prefix
-		constraint = substituteUVMPath(sandboxID, constraint)
-		if err = constraint.validate(m); err == nil {
-			return nil
-		}
+// createAllowAllEnforcer creates and returns OpenDoorSecurityPolicyEnforcer
+// instance.  The provided base64EncodedPolicy must be either empty or contain
+// exactly the field "allow_all" set to true.
```

Comment:
"exactly the field "allow_all" set to true" We don't want that anymore. They can use a long hand allow all policy.